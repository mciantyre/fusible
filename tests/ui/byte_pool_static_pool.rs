// SPDX-License-Identifier: MPL-2.0
// SPDX-FileCopyrightText: Copyright 2024 Ian McIntyre

use fusible::byte_pool::{BytePool, BytePoolContext};

static BYTE_POOL: BytePoolContext = BytePool::context();

/// A static byte pool cannot be created with local storage.
fn main() {
    let mut storage = [0u8; 1024];
    let pool = core::pin::Pin::static_ref(&BYTE_POOL);
    BytePool::create(pool, &mut storage, &Default::default()).unwrap();
}
