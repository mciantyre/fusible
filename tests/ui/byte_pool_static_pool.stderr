error[E0597]: `storage` does not live long enough
  --> tests/ui/byte_pool_static_pool.rs:12:28
   |
10 |     let mut storage = [0u8; 1024];
   |         ----------- binding `storage` declared here
11 |     let pool = core::pin::Pin::static_ref(&BYTE_POOL);
   |                -------------------------------------- assignment requires that `storage` is borrowed for `'static`
12 |     BytePool::create(pool, &mut storage, &Default::default()).unwrap();
   |                            ^^^^^^^^^^^^ borrowed value does not live long enough
13 | }
   | - `storage` dropped here while still borrowed
